src/compas/scene/scene.py

```diff
@@ -6,58 +6,241 @@
 from .context import clear
 
 
+class SceneObjectNode(TreeNode):
+    """A node representing a scene object in a scene tree. The SceneObjectNode should only be used internally by the SceneTree.
+
+    Parameters
+    ----------
+    object : :class:`compas.scene.SceneObject`
+        The scene object associated with the node.
+
+    Attributes
+    ----------
+    object : :class:`compas.scene.SceneObject`
+        The scene object associated with the node.
+    parent_object : :class:`compas.scene.SceneObject`
+        The scene object associated with the parent node.
+    child_objects : list[:class:`compas.scene.SceneObject`]
+        The scene objects associated with the child nodes.
+
+    """
+
+    def __init__(self, sceneobject):
+        super(SceneObjectNode, self).__init__(name=sceneobject.name)
+        self.object = sceneobject
+
+    @property
+    def parent_object(self):
+        if self.parent and isinstance(self.parent, SceneObjectNode):
+            return self.parent.object
+        return None
+
+    @property
+    def child_objects(self):
```

Comment:
same here, even though "child" is not a verb. for consistency i would name this ``childobjects``